plugins/inx/server_utxo.go

```diff
@@ -216,10 +233,92 @@ func (s *INXServer) ListenToLedgerUpdates(req *inx.LedgerUpdateRequest, srv inx.
 	return ctx.Err()
 }
 
+func (s *INXServer) ListenToTreasuryUpdates(req *inx.LedgerRequest, srv inx.INX_ListenToTreasuryUpdatesServer) error {
+	var sentTreasuryUpdate bool
+	sendPreviousMilestoneDiffs := func(startIndex milestone.Index) error {
+		deps.UTXOManager.ReadLockLedger()
+		defer deps.UTXOManager.ReadUnlockLedger()
+
+		ledgerIndex, err := deps.UTXOManager.ReadLedgerIndex()
+
+		if startIndex > 0 {
+			// Stream all available milestone diffs first
+			pruningIndex := deps.Storage.SnapshotInfo().PruningIndex
+			if startIndex <= pruningIndex {
+				return status.Errorf(codes.InvalidArgument, "given startMilestoneIndex %d is older than the current pruningIndex %d", startIndex, pruningIndex)
+			}
+
+			if err != nil {
```

Comment:
I think that code should be moved up to `ledgerIndex, err := deps.UTXOManager.ReadLedgerIndex()`